x-pack/solutions/security/packages/expandable-flyout/src/hooks/use_window_width.ts

```diff
@@ -33,46 +40,79 @@ export const useWindowWidth = (): number => {
-            (RIGHT_SECTION_MAX_WIDTH - RIGHT_SECTION_MIN_WIDTH) *
+          const ratioWidthOverlayMode =
+            (RIGHT_SECTION_MAX_WIDTH_OVERLAY_MODE - RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE) *
             ((windowWidth - MIN_RESOLUTION_BREAKPOINT) /
               (MAX_RESOLUTION_BREAKPOINT - MIN_RESOLUTION_BREAKPOINT));
 
           // the right section's width will grow to 750px (at 1920px resolution) and will never go bigger than 750px in higher resolutions
-          rightSectionWidth = Math.min(
-            RIGHT_SECTION_MIN_WIDTH + ratioWidth,
-            RIGHT_SECTION_MAX_WIDTH
+          rightSectionWidthOverlayMode = Math.min(
+            RIGHT_SECTION_MIN_WIDTH_OVERLAY_MODE + ratioWidthOverlayMode,
+            RIGHT_SECTION_MAX_WIDTH_OVERLAY_MODE
+          );
+
+          const ratioWidthPushMode =
+            (RIGHT_SECTION_MAX_WIDTH_PUSH_MODE - RIGHT_SECTION_MIN_WIDTH_PUSH_MODE) *
+            ((windowWidth - FULL_WIDTH_BREAKPOINT) /
+              (BIG_RESOLUTION_BREAKPOINT - FULL_WIDTH_BREAKPOINT));
+
+          // the right section's width will grow to 750px (at 1920px resolution) and will never go bigger than 750px in higher resolutions
+          rightSectionWidthPushMode = Math.min(
+            RIGHT_SECTION_MIN_WIDTH_PUSH_MODE + ratioWidthPushMode,
+            RIGHT_SECTION_MAX_WIDTH_PUSH_MODE
           );
         }
 
-        let leftSectionWidth: number;
+        let leftSectionWidthOverlayMode: number;
         // the left section's width will be nearly the remaining space for resolution lower than 1600px
         if (windowWidth <= FULL_WIDTH_BREAKPOINT) {
-          leftSectionWidth = windowWidth - rightSectionWidth - FULL_WIDTH_PADDING;
+          leftSectionWidthOverlayMode =
+            windowWidth - rightSectionWidthOverlayMode - FULL_WIDTH_PADDING;
         } else {
           // the left section's width will be taking 80% of the remaining space for resolution higher than 1600px, while never going bigger than 1500px
-          leftSectionWidth = Math.min(
-            ((windowWidth - rightSectionWidth) * 80) / 100,
+          leftSectionWidthOverlayMode = Math.min(
+            ((windowWidth - rightSectionWidthOverlayMode) * 80) / 100,
             LEFT_SECTION_MAX_WIDTH
           );
         }
 
-        const previewSectionWidth: number = rightSectionWidth;
+        const ratioWidthPushMode =
+          (LEFT_SECTION_MAX_WIDTH_PUSH_MODE - LEFT_SECTION_MIN_WIDTH_PUSH_MODE) *
+          ((windowWidth - FULL_WIDTH_BREAKPOINT) /
+            (BIG_RESOLUTION_BREAKPOINT - FULL_WIDTH_BREAKPOINT));
+
+        // the right section's width will grow to 750px (at 1920px resolution) and will never go bigger than 750px in higher resolutions
+        const leftSectionWidthPushMode = Math.min(
+          LEFT_SECTION_MIN_WIDTH_PUSH_MODE + ratioWidthPushMode,
+          LEFT_SECTION_MAX_WIDTH_PUSH_MODE
+        );
+
+        const previewSectionWidthOverlayMode: number = rightSectionWidthOverlayMode;
```

Comment:
is there value in storing previewWidth, if it is always the same as right section width?

Reply:
I had the same though months ago when we first implemented this and there was only `overlay` mode. I believed then it would be clear when looking at the store (in the code or in Redux Dev tools for example) what would be the value of the preview width.
I still feel the same. We could indeed remove that value, but having it there has basically no downside in terms of performance or storage used... for me it's more when I debug things, I can easily see the calculated value in my Redux Dev tools and I don't have to remember that it would be the same as the right section and why...

Reply:
sounds good to me. one caveat is that we just need to make sure preview is used consistently, I made 2 comments in calculating the preview percentage

Reply:
yup, I updated those 2 places to use the preview values